### DISCUSSION TITLE
Wanted your thoughts on a new search plugin for Obsidian called Khoj

### DISCUSSION THREAD

**Discussion by sohanglal** [scambier/obsidian-omnisearch#192] [Ideas]
I wanted to know what y'all think about [Khoj](https://github.com/debanjum/khoj)?
The front-end of the plugin is not as refined as Omnisearch but I think the search functionality of Khoj might be very powerful. 
Maybe there could be some collaboration between Khoj and Omnisearch.

**Comment by scambier** [comment]
Definitely looks like an interesting project, and Omnisearch could theoretically (and optionally) use Khoj's backend... I'll keep an eye on it and add it to my list of "things to try" 😅

**Reply by debanjum** [reply]
Hi @scambier, author of Khoj here 👋🏾. 

I'd be open to having Khoj integrate with Omnisearch. Let me know when/if you get to this. I'd like to have the Khoj backend API versioned before having downstream dependencies to avoid chance of breakage

**Reply by scambier** [reply]
No pressure, there's nothing yet in Omnisearch to consume a different backend and I'm kinda busy on other projects right now :) In any case I'll wait until you're confident enough in your API to publish something.

I saw your Reddit post too. If you'd like some help to integrate OCR and PDFs for Obsidian, I specifically developed https://github.com/scambier/obsidian-text-extractor#using-text-extractor-as-a-dependency-for-your-plugin to be used as an agnostic dependency. It's still a bit rough (especially on the PDF side), but there's a basic API that Kohj could use if you want to try it.

**Reply by debanjum** [reply]
> No pressure, there's nothing yet in Omnisearch to consume a different backend and I'm kinda busy on other projects right now :) 

That's good to know 👍🏽 

> In any case I'll wait until you're confident enough in your API to publish something.

The API should be mature enough. Just haven't versioned it yet, as no downstream dependencies had been planned to consume it.

> I specifically developed https://github.com/scambier/obsidian-text-extractor#using-text-extractor-as-a-dependency-for-your-plugin to be used as an agnostic dependency. It's still a bit rough (especially on the PDF side), but there's a basic API that Kohj could use if you want to try it.

Your text-extractor plugin looks neat! But it may not be the best fit for Khoj. Khoj uses a server-client architecture. And most of the search, content indexing logic is on the Python backend. So a Python based text extractor would be better for Khoj
